🔔🔋 Battery Status & Level Alert Enhancements

AlexLemus-Dev opened the pull request:
### **🔋🎵 Brief summary:**

The battery notification system was enhanced, allowing users to configure both the battery thresholds for alerts and the custom sounds for each notification type.

### **📝 Actions performed:**

- 🔋 Added settings for low/high battery notifications (user can choose the % threshold).
- 🎵 Introduced the ability to select system sounds for power status, low battery, and high battery notifications.
- 🖥️ Updated SettingsView to allow users to pick sounds for each notification type.
- 🛠️ Updated SettingsView with new pickers for configuring thresholds.
- ⚡ Improved BatteryStatusViewModel to handle customizable battery level thresholds.
- ⚡ Extended BatteryStatusViewModel and Constants to support and trigger these sound alerts.

<img width="722" height="625" alt="Screenshot 2025-09-13 at 10 26 17 a m" src="https://github.com/user-attachments/assets/35e8acc2-ff84-4aaa-ad65-5cea251b7298" />

AlexLemus-Dev committed:
🔔🔋 Battery Status & Level Alert Enhancements

- 🔋 Added settings for low/high battery notifications (user can choose the % threshold).
- 🎵 Introduced the ability to select system sounds for power status, low battery, and high battery notifications.
- 🖥️ Updated SettingsView to allow users to pick sounds for each notification type.
- 🛠️ Updated SettingsView with new pickers for configuring thresholds.
- ⚡ Improved BatteryStatusViewModel to handle customizable battery level thresholds.
- ⚡ Extended BatteryStatusViewModel and Constants to support and trigger these sound alerts.

AlexLemus-Dev committed:
Merge branch 'dev' into feature/battery-threshold-notifications

Alexander5015 commented:
Sorry about the delay in getting to this. It looks great! Can you fix the conflicts?

AlexLemus-Dev committed:
Merge branch 'dev' into feature/battery-threshold-notifications

Alexander5015 commented on boringNotch/components/Settings/SettingsView.swift:
```diff
@@ -522,12 +589,8 @@ struct HUD: View {
-                Defaults.Toggle(key: .systemEventIndicatorUseAccent) {
-                    Text("Use accent color")
-                }
+                Defaults.Toggle("Enable glowing effect", key: .systemEventIndicatorShadow)
```
```suggestion
                Defaults.Toggle(key: .systemEventIndicatorShadow) {
                    Text("Enable glowing effect")
                }
```

AlexLemus-Dev commented on boringNotch/components/Settings/SettingsView.swift:
```diff
@@ -343,27 +341,176 @@ struct GeneralSettings: View {
+    
+}
+
+struct Downloads: View {
```
Oops, my bad — I’m not sure when that got uncommented. I might’ve triggered a shortcut by accident, but I’ve already fixed it.

AlexLemus-Dev committed:
Update boringNotch/components/Settings/SettingsView.swift

Co-authored-by: Alexander <33609792+Alexander5015@users.noreply.github.com>

AlexLemus-Dev committed:
Update boringNotch/components/Settings/SettingsView.swift

Co-authored-by: Alexander <33609792+Alexander5015@users.noreply.github.com>

AlexLemus-Dev committed:
Update boringNotch/components/Settings/SettingsView.swift

Co-authored-by: Alexander <33609792+Alexander5015@users.noreply.github.com>

AlexLemus-Dev committed:
Update boringNotch/components/Settings/SettingsView.swift

Co-authored-by: Alexander <33609792+Alexander5015@users.noreply.github.com>

AlexLemus-Dev committed:
Update boringNotch/components/Settings/SettingsView.swift

Co-authored-by: Alexander <33609792+Alexander5015@users.noreply.github.com>

AlexLemus-Dev committed:
Update boringNotch/components/Settings/SettingsView.swift

Co-authored-by: Alexander <33609792+Alexander5015@users.noreply.github.com>

AlexLemus-Dev committed:
Update boringNotch/components/Settings/SettingsView.swift

Co-authored-by: Alexander <33609792+Alexander5015@users.noreply.github.com>

AlexLemus-Dev committed:
fix(Localizable): Fixed new change localizables

AlexLemus-Dev committed:
Merge branch 'dev' into feature/battery-threshold-notifications

AlexLemus-Dev committed:
Merge branch 'dev' into feature/battery-threshold-notifications

Alexander5015 reviewed (commented):
Thanks for the update! It looks like the build is currently failing. Could you review the CI logs and address the errors?

AlexLemus-Dev committed:
fix(SettingsView): misplaced closing brace in Charge view

Moved the closing brace for the header section in the Charge view to ensure proper SwiftUI section structure.

AlexLemus-Dev committed:
Merge branch 'dev' into feature/battery-threshold-notifications

AlexLemus-Dev committed:
Merge branch 'dev' into feature/battery-threshold-notifications

AlexLemus-Dev committed:
Merge branch 'dev' into feature/battery-threshold-notifications

Alexander5015 reviewed (changes requested):
Its still not building properly

AlexLemus-Dev committed:
fix(Consntans): Remove duplicated constant showPowerStatusNotifications

Eliminated the duplicate showPowerStatusNotifications constant to clean up the file.

AlexLemus-Dev commented:
> Its still not building properly

I fixed the issue — the project wasn’t building because the showPowerStatusNotifications constant was duplicated.

AlexLemus-Dev committed:
Merge branch 'dev' into feature/battery-threshold-notifications

AlexLemus-Dev committed:
Merge branch 'dev' into feature/battery-threshold-notifications

Alexander5015 commented on boringNotch/Localizable.xcstrings:
```diff
@@ -4543,7 +4543,7 @@
           "stringUnit" : {
             "state" : "needs_review",
-            "value" : "Charge en pause : mode bureau"
+            "value" : "Charge en pause : mode bureau"
```
Localization file should be unchanged

Alexander5015 commented on boringNotch/components/Settings/SettingsView.swift:
```diff
@@ -349,37 +349,109 @@ struct GeneralSettings: View {
-                Defaults.Toggle(key: .showPowerStatusNotifications) {
-                    Text("Show power status notifications")
-                }
+                Defaults.Toggle("Show battery indicator", key: .showBatteryIndicator)
```
Why did you make this change?

Alexander5015 commented on boringNotch/components/Settings/SettingsView.swift:
```diff
@@ -349,37 +349,109 @@ struct GeneralSettings: View {
+                Defaults.Toggle("Show power status icons", key: .showPowerStatusIcons)
             } header: {
-                Text("Battery Information")
+                Text("Information")
```
Why was this renamed?

Alexander5015 commented on boringNotch/models/BatteryStatusViewModel.swift:
```diff
@@ -78,13 +91,13 @@ class BatteryStatusViewModel: ObservableObject {
-                    ? "Charging battery"
-                    : (self.levelBattery < self.maxCapacity ? "Not charging" : "Full charge")
+                    ? "Charging"
+                    : (self.levelBattery < self.maxCapacity ? "Not Charging" : "Full Charge")
```
Can you help me understand why these were updated? I thought they were fine as-is.

Alexander5015 commented on boringNotch/models/BatteryStatusViewModel.swift:
```diff
@@ -116,17 +130,47 @@ class BatteryStatusViewModel: ObservableObject {
             self.statusText = batteryInfo.isPluggedIn ? "Plugged In" : "Unplugged"
         }
+        Task {
+            try? await Task.sleep(for: .seconds(2.0))
```
Why is there a 2 second sleep here?

Alexander5015 reviewed (changes requested):
Left a few comments for you to check out. btw, there are some merge conflicts to sort out as well 🙃
